airflow/models/taskinstance.py

```diff
@@ -968,6 +968,7 @@ def _run_raw_task(
                 context = self.get_template_context()
 
                 task_copy = copy.copy(task)
+                task_copy.lock_for_execution()
```

Comment:
How about we combine both these in to a single function `task.prepare_for_execution()` that returns a new copied instance with the "lock" set.

We also do `task_copy = copy.copy(task) in the `dry_run` fn later on -- we should use that here to.

Reply:
Awesome idea! I was thinking about adding this lock somewhere but nothing came to my mind